src/AdoNetCore.AseClient/Internal/SchemaTableBuilder.cs

```diff
@@ -163,28 +168,45 @@ private void TryLoadKeyInfo(DataTable table, string baseTableNameValue, string b
 
                 try
                 {
+                    var columnMetadata = new Dictionary<string, ColumnMetadata>();
                     using (var keyInfoDataReader = command.ExecuteReader())
                     {
                         while (keyInfoDataReader.Read())
                         {
-                            var keyColumnName = keyInfoDataReader["COLUMN_NAME"].ToString();
-                            var keySchemaName = keyInfoDataReader["TABLE_SCHEMA"].ToString();
-                            var keyCatalogName = keyInfoDataReader["TABLE_CATALOG"].ToString();
+                            var key = keyInfoDataReader["COLUMN_NAME"].ToString();
+
+                            // Check for duplicate name and if we found any then give it a miss
+                            if (columnMetadata.ContainsKey(key))
+                            {
+                                keyInfoDataReader.Close();
+                                return;
```

Comment:
Surely you want to continue here, not return?

Reply:
If we fell into this situation, I felt it's not safe to carry on setting the row "IsKey" and "IsUnique"  values and hence keyInfoDataReader.Close(); was called prior too.

Reply:
Ah, that explains why you split the loops. By the way, you don't need to close the reader since it's created in a using statement. 

What does the reference driver do in this scenario, or does it not retrieve the metadata in the same way?

Reply:
I don't think the reference driver retrieve the metadata the same way. I can't see anything in the reference TDS message regarding calling out the this metadata stored proc